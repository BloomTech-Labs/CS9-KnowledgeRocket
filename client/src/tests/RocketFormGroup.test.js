import { renderSnapshotTest } from './utils';
import 'jest-styled-components';
// Component
import {
    Blurb,
    ErrorText,
    FormGroup,
    QuestionChoices,
    QuestionWrapper,
    TextArea,
    errorHelper,
    generateErrorIdFrom,
} from '../components/Rocket/FormGroup';

// simple styled components
const Grouplets = {
    Blurb,
    ErrorText,
    FormGroup,
    QuestionWrapper,
    TextArea,
};

describe('RocketFormGroup', () => {
    describe('testing styled components', () => {
        describe('render', () => {
            for (const ComponentName in Grouplets) {
                if (Grouplets.hasOwnProperty(ComponentName)) {
                    const Grouplet = Grouplets[ComponentName];
                    describe(`<${ComponentName} />`, () => {
                        renderSnapshotTest(Grouplet);
                    });
                }
            }
        });
    });
    describe('<QuestionChoices />', () => {
        const mockProps = {
            values: {
                tm: {
                    explanation: '',
                    choices: [
                        {
                            text: '',
                        },
                        {
                            text: '',
                        },
                        {
                            text: '',
                        },
                        {
                            text: '',
                        },
                    ],
                },
            },

            interval: 'tm',

            errors: {
                tm: '',
            },

            touched: {},
        };
        renderSnapshotTest(QuestionChoices, mockProps);
        // TODO prop tests
    });

    describe('errorHelper', () => {
        it('displays correct error', () => {
            const mockError = {
                td: 'I errored',
            };
            const mockTouch = {
                td: true,
            };
            const mockProperty = 'td';

            const mockDisplayError = errorHelper(mockError, mockTouch);
            expect(mockDisplayError(mockProperty)).toBe(mockError.td);
        });
        it('does not return error when value is untouched', () => {
            const mockError = {
                td: 'I errored',
            };
            const mockTouch = {};
            const mockProperty = 'td';

            const mockDisplayError = errorHelper(mockError, mockTouch);
            expect(mockDisplayError(mockProperty)).toBe(undefined);
        });
    });
    describe('generateIdFrom', () => {
        it('correctly generates id', () => {
            const attr = 'td';
            const identifier = 'question';
            const expected = `${attr}-${identifier}-ErrorDescription`;
            expect(generateErrorIdFrom(attr, identifier)).toBe(expected);
        });
    });
});
