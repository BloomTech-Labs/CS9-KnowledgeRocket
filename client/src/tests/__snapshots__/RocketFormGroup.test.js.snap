// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketFormGroup <QuestionChoices /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  width: 100%;
  height: 4rem;
}

.c3 {
  display: grid;
  margin: 0 0 1rem 0;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  width: 100%;
}

.c2 {
  color: red;
  font-size: .8rem;
  height: 1rem;
}

<section>
  <div
    className="c0"
  >
    <label
      htmlFor="tm.explanation"
    >
      Two Months - Review Text
    </label>
    <textarea
      aria-describedby="tm-explanation-ErrorDescription"
      className="c1"
      name="tm.explanation"
      onChange={undefined}
      placeholder="Two Day Review Text"
      value=""
    />
    <span
      className="c2"
      id="tm-explanation-ErrorDescription"
    />
  </div>
  <div
    className="c0"
  >
    <label
      htmlFor="tm.question"
    >
      Two Months - Review Question
    </label>
    <textarea
      aria-describedby="tm-question-ErrorDescription"
      className="c1"
      name="tm.question"
      onChange={undefined}
      placeholder="Two Day Review Text"
      value={undefined}
    />
    <span
      className="c2"
      id="tm-question-ErrorDescription"
    />
  </div>
  <div
    className="c3"
  >
    <label>
      <input
        aria-describedby="tm-correct-ErrorDescription"
        checked={false}
        name="tm.correct"
        onBlur={undefined}
        onChange={undefined}
        type="radio"
        value=""
      />
      <input
        name="tm.choices[0].text"
        onChange={undefined}
        type="text"
        value=""
      />
    </label>
    <label>
      <input
        aria-describedby="tm-correct-ErrorDescription"
        checked={false}
        name="tm.correct"
        onBlur={undefined}
        onChange={undefined}
        type="radio"
        value=""
      />
      <input
        name="tm.choices[1].text"
        onChange={undefined}
        type="text"
        value=""
      />
    </label>
    <label>
      <input
        aria-describedby="tm-correct-ErrorDescription"
        checked={false}
        name="tm.correct"
        onBlur={undefined}
        onChange={undefined}
        type="radio"
        value=""
      />
      <input
        name="tm.choices[2].text"
        onChange={undefined}
        type="text"
        value=""
      />
    </label>
    <label>
      <input
        aria-describedby="tm-correct-ErrorDescription"
        checked={false}
        name="tm.correct"
        onBlur={undefined}
        onChange={undefined}
        type="radio"
        value=""
      />
      <input
        name="tm.choices[3].text"
        onChange={undefined}
        type="text"
        value=""
      />
    </label>
  </div>
  <span
    className="c2"
    id="tm-correct-ErrorDescription"
  />
</section>
`;

exports[`RocketFormGroup testing styled components render <Blurb /> renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 4rem;
  margin: 1rem 0;
  text-align: justify;
}

<p
  className="c0"
/>
`;

exports[`RocketFormGroup testing styled components render <ErrorText /> renders correctly 1`] = `
.c0 {
  color: red;
  font-size: .8rem;
  height: 1rem;
}

<span
  className="c0"
/>
`;

exports[`RocketFormGroup testing styled components render <FormGroup /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

<div
  className="c0"
/>
`;

exports[`RocketFormGroup testing styled components render <QuestionWrapper /> renders correctly 1`] = `
.c0 {
  display: grid;
  margin: 0 0 1rem 0;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  width: 100%;
}

<div
  className="c0"
/>
`;

exports[`RocketFormGroup testing styled components render <TextArea /> renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 4rem;
}

<textarea
  className="c0"
/>
`;
